src/measure_theory/measure/haar_of_basis.lean

```diff
@@ -0,0 +1,159 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+
+import measure_theory.measure.haar
+import analysis.inner_product_space.pi_L2
+
+/-!
+# Additive Haar measure constructed from a basis
+
+Given a basis of a finite-dimensional real vector space, we define the corresponding Lebesgue
+measure, which gives measure `1` to the parallelogram spanned by the basis.
+
+## Main definitions
+
+* `parallelogram v` is the parallelogram spanned by a finite family of vectors.
+* `basis.parallelogram` is the parallelogram associated to a basis, seen as a compact set with
+nonempty interior.
+* `basis.add_haar` is the Lebesgue measure associated to a basis, giving measure `1` to the
+corresponding parallelogram.
+
+In particular, we declare a `measure_space` instance on any finite-dimensional inner product space,
+by using the Lebesgue measure associated to some orthonormal basis (which is in fact independent
+of the basis).
+-/
+
+open set topological_space measure_theory measure_theory.measure finite_dimensional
+open_locale big_operators
+
+noncomputable theory
+
+variables {ι ι' E F : Type*} [fintype ι] [fintype ι']
+
+section add_comm_group
+
+variables [add_comm_group E] [module ℝ E] [add_comm_group F] [module ℝ F]
+
+/-- The parallelogram spanned by a finite family of vectors. -/
```

Comment:
Perhaps worth mentioning in the docstring that it includes all the boundaries; I wouldn't be surprised to see an alternative definition with `Ico`.

Including the word "parallelepiped" in the docstring for the sake of searchability is probably a good idea too.

Reply:
Yes, in fact I wasn't completely sure of the name parallelogram here. I will probably switch to parallelepiped, which is less two-dimensional.